hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/client/DefaultCertificateClient.java

```diff
@@ -922,7 +922,7 @@ public Duration timeBeforeExpiryGracePeriod(X509Certificate certificate) {
     Duration gracePeriod = securityConfig.getRenewalGracePeriod();
     Date expireDate = certificate.getNotAfter();
     LocalDateTime gracePeriodStart = expireDate.toInstant()
-        .atZone(ZoneId.systemDefault()).toLocalDateTime().minus(gracePeriod);
+        .minus(gracePeriod).atZone(ZoneId.systemDefault()).toLocalDateTime();
```

Comment:
Same logic appears in `CertificateClientTestImpl`, should be fixed there, too?

https://github.com/apache/ozone/blob/111161d78253e2afe920b14d2a14be282ea57e90/hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/security/x509/certificate/client/CertificateClientTestImpl.java#L158-L159

Reply:
Yes, thank you @adoroszlai for spotting this one.
In general this one did not caused a problem, as this implementation is used with functionality that does not happen around certificate renewal, and where it does, the grace period is a few seconds, so if it remains this way it does not cause trouble just for very short periods of time, so probably we would not recognize it, but nonetheless, I have pushed the change to the test class in a follow up commit.